src/Template.php

```diff
@@ -689,112 +632,86 @@ public function loadTemplateFromString($str)
 
     /**
      * Used for adding unique tag alternatives. E.g. if your template has
-     * {$name}{$name}, then first would become 'name#1' and second 'name#2', but
+     * {$name}{$name}, then first would become 'name#0' and second 'name#1', but
      * both would still respond to 'name' tag.
      *
      * @var array
      */
     private $tag_cnt = [];
 
     /**
-     * Register tags and return unique tag name.
-     *
-     * @param string $tag tag name
-     *
-     * @return string unique tag name
+     * Register tag and return unique tag name.
      */
-    protected function regTag($tag)
+    protected function regTag(string $tag): string
     {
         if (!isset($this->tag_cnt[$tag])) {
-            $this->tag_cnt[$tag] = 0;
+            $this->tag_cnt[$tag] = -1;
```

Comment:
could we keep those at 0 and start tags with #1? may introduce a minor compatibility issue. I have used positional tag references in the past (although long time ago).

Reply:
rebuildTags started on #0 already and we normalize it now to the more programmer way

Reply:
closing this as this is now internal representaton only